ecl/hql/hqlgram2.cpp

```diff
@@ -2270,17 +2270,23 @@ void HqlGram::checkAllAssigned(IHqlExpression * originalRecord, IHqlExpression *
 
 void HqlGram::checkFoldConstant(attribute & attr)
 {
-    Owned<IHqlExpression> expr = attr.getExpr();
-    attr.setExpr(foldHqlExpression(expr), attr);
-
-    checkConstant(attr);
+    if (attr.queryExpr()->isConstant())
+    {
+        Owned<IHqlExpression> expr = attr.getExpr();
+        attr.setExpr(foldHqlExpression(expr), attr);
+    }
+    else
+        checkConstant(attr);
 }
 
 IHqlExpression * HqlGram::checkConstant(const attribute & errpos, IHqlExpression * expr)
```

Comment:
@richardkchapman I think this provokes a general question we should discuss in the sprint/team meeting.  When should documentation be added to a function/class?

- When a new function or class is added?  I think we all agree yes.
- When a function is modified?  (The case for checkFoldConstant in this situation)
- When new code refers to an existing function?  (Vaguely the case in this situation for checkConstant - there are no new calls to the function.)

Arguments for always adding it are that it is the only way that the system is going to be documented.  If everyone knows that is what is expected then we will get into the habit of it, but it needs to be applied consistently.

Requiring any function that is changed (checkFoldConstant) to be documented would be reasonable - but do we want to make it a requirement?  Requiring documentation on anything that is called from modified code is likely to be too much.

There is also the additional problem if you modify code in a function you don't fully understand - but then maybe you shouldn't be modifying it until you fully understand it!